cmd/query/app/querysvc/adjuster/spaniduniquifier.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019 The Jaeger Authors.
+// Copyright (c) 2017 Uber Technologies, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package adjuster
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	warningTooManySpans = "cannot assign unique span ID, too many spans in the trace"
+)
+
+// SpanIDUniquifier returns an adjuster that changes span ids for server
+// spans (i.e. spans with tag: span.kind == server) if there is another
+// client span that shares the same span ID. This is needed to deal with
+// Zipkin-style clients that reuse the same span ID for both client and server
+// side of an RPC call. Jaeger UI expects all spans to have unique IDs.
+//
+// This adjuster never returns any errors. Instead it records any issues
+// it encounters in Span.Warnings.
+func SpanIDUniquifier() *SpanIDDeduper {
+	return &SpanIDDeduper{}
+}
+
+type SpanIDDeduper struct {
+	spansByID map[pcommon.SpanID][]ptrace.Span
```

Comment:
Doesn't this make the struct use-once? The StandardAdjusters list is only instantiated once, so the same adjuster instance would be used for many traces.

Reply:
@yurishkuro yeah it does - was just trying to get the implementation working. i'll fix this